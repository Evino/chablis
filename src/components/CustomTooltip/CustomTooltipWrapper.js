import styled from 'styled-components';

import TooltipWrapper from '../Tooltip/TooltipWrapper';

const CustomTooltipWrapper = styled(TooltipWrapper)`
  position: absolute;
  min-width: 255px;
  max-width: 280px;
  padding: 16px 24px;
  ${({ alignY }) => alignY}: calc(100% + 15px);
  ${({ alignX }) => alignX}: ${({ spaced }) => (spaced ? '-55px' : 'calc(50% - 32px)')};

  img {
    width: 100%;
    height: auto;
    margin-bottom: 16px;
    pointer-events: none;
  }

  p {
    color: #696969;
    font-size: 0.75rem;
    line-height: 0.875rem;
    text-align: left;
  }
`;

export default CustomTooltipWrapper;
