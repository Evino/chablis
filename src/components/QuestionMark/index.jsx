import styled from 'styled-components';

import {
  colors,
  fontSizes,
  fontWeights,
  sizes,
} from '../../styles/variables';

const QuestionMark = styled.button`
  position: relative;
  display: inline-flex;
  align-items: center;
  justify-content: center;
  width: ${sizes.DEFAULT};
  height: ${sizes.DEFAULT};
  background-color: ${colors.GREEN};
  border-width: 0;
  border-radius: 50%;
  padding: 0;
  margin: 0;
  cursor: help;
  transition: opacity .2s ease-out;

  :hover,
  :focus {
    opacity: .8;
    outline: none;
  }

  :active {
    opacity: .7;
  }

  > span {
    :first-of-type {
      display: inline-block;
      text-indent: -99999px;
    }
  }

  ::before {
    content: '?';
    color: ${colors.WHITE};
    font-size: ${fontSizes.SMALL};
    font-weight: ${fontWeights.NORMAL};
    line-height: ${fontSizes.NORMAL};
  }
`;

export default QuestionMark;
