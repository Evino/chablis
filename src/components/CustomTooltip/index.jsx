import React from 'react';
import PropTypes from 'prop-types';

import CustomTooltipWrapper from './CustomTooltipWrapper';
import { GetPosition } from '../../Utils';

const CustomTooltip = ({
  align,
  children,
  spaced,
  show,
  type,
}) => {
  const position = GetPosition(align);

  const CustomTooltipProps = {
    alignX: position[0],
    alignY: position[1],
    show,
    spaced,
    type,
  };

  return <CustomTooltipWrapper {...CustomTooltipProps}>{children}</CustomTooltipWrapper>;
};

CustomTooltip.propTypes = {
  align: PropTypes.oneOf([
    'bottom-start',
    'bottom-end',
    'top-start',
    'top-end',
  ]),
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]).isRequired,
  spaced: PropTypes.bool,
  show: PropTypes.bool,
  type: PropTypes.oneOf([
    'default',
    'success',
    'danger',
  ]),
};

CustomTooltip.defaultProps = {
  align: 'top-end',
  spaced: false,
  show: true,
  type: 'default',
};

export default CustomTooltip;
