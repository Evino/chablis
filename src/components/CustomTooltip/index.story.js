import React from 'react';
import { storiesOf } from '@storybook/react';

import CustomTooltip from '.';

const divStyle = {
  marginTop: '30px',
  marginLeft: '100px',
};

const buttonStyle = {
  position: 'relative',
};

const ImageURL = 'https://evino-res.cloudinary.com/image/upload/v1614018200/web/assets/escala.png';

const CustomTooltipWithImage = (
  <CustomTooltip type="default" show align="bottom-start">
    <img src={ImageURL} alt="Escala" />
    <p>
      Classificação de acordo com a legislação brasileira em relação ao
      conteúdo de açúcar, expresso em gramas por litro de glicose (g/L).
    </p>
  </CustomTooltip>
);

storiesOf('Custom Tooltip', module).addWithInfo('Default', () => (
  <div style={divStyle}>
    <button type="button" style={buttonStyle}>
      Click me!
      {CustomTooltipWithImage}
    </button>
  </div>
));
