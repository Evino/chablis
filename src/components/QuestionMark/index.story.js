import React from 'react';
import { storiesOf } from '@storybook/react';
import QuestionMark from '.';

const divStyle = {
  marginTop: '30px',
  marginLeft: '300px',
};

storiesOf('QuestionMark', module).addWithInfo('Default', () => (
  <div style={divStyle}>
    <QuestionMark />
  </div>
));
