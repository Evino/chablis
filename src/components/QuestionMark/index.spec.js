import React from 'react';
import renderer from 'react-test-renderer';
import QuestionMark from '.';
import CustomTooltip from '../CustomTooltip';

const element = <QuestionMark />;

const elementWithText = (
  <QuestionMark>
    <span>Saiba mais</span>
  </QuestionMark>
);

const elementWithCustomTooltip = (
  <QuestionMark>
    <span>Saiba mais</span>
    <CustomTooltip>
      <p>This is a custom tooltip</p>
    </CustomTooltip>
  </QuestionMark>
);

describe('Tooltip', () => {
  it('should render a tooltip', () => {
    const tree = renderer.create(element).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('Shloud render with accessibility text', () => {
    const tree = renderer.create(elementWithText).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('Shloud render with Custom tooltip element inside', () => {
    const tree = renderer.create(elementWithCustomTooltip).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
