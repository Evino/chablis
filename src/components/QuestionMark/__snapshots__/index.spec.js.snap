// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip Shloud render with Custom tooltip element inside 1`] = `
<button
  className="sc-bdVaJa cnVwij"
>
  <span>
    Saiba mais
  </span>
  <span
    className="sc-htpNat AtctT sc-bwzfXH dlWQTx"
    type="default"
  >
    <p>
      This is a custom tooltip
    </p>
  </span>
</button>
`;

exports[`Tooltip Shloud render with accessibility text 1`] = `
<button
  className="sc-bdVaJa cnVwij"
>
  <span>
    Saiba mais
  </span>
</button>
`;

exports[`Tooltip should render a tooltip 1`] = `
<button
  className="sc-bdVaJa cnVwij"
/>
`;
